src/main/java/tasks/Deadline.java

```diff
@@ -0,0 +1,66 @@
+package tasks;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import duke.DukeException;
+
+public class Deadline extends Task {
+    private String[] dateAndTime;
+    private LocalDate date;
+    private String time;
+
+    /**
+     * Creates a Deadline task and sets its date and time accordingly from user input.
+     *
+     * @param description Description of the deadline task generated from user input.
+     * @throws DukeException If the input format is not followed or there are missing information.
+     */
+    public Deadline(String description) throws DukeException {
+        super(description);
+        try {
+            String[] strArr = description.split("/by ");
+            this.description = strArr[0];
+            dateAndTime = strArr[1].split(" ");
+            date = LocalDate.parse(dateAndTime[0]);
+            time = dateAndTime[1];
+            super.saveFormat = "D," + this.description + "," + date + "," + time + "," + super.isDone;
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new DukeException();
+        }
+    }
+
+    /**
+     * Creates a Deadline task from previously saved list of tasks during initialisation of chat-bot.
```

Comment:
I like the level of description in the JavaDoc comments for the methods! Perhaps you could also consider using @link to reference another member of the java classes!